package.json

```diff
@@ -34,6 +36,7 @@
     "test-examples": "npm run build && npm run test-example:legacy && npm run test-example:flat",
     "test-example:legacy": "cd examples/legacy && npm install && npm run lint",
     "test-example:flat": "cd examples/flat && npm install && npm run lint",
+    "test-types": "npx --package typescript@latest tsc --noEmit index.d.ts",
```

Comment:
shouldn't noEmit be set in tsconfig, and tsc be run on the whole project?

also, let's add typescript to the dev deps rather than using npx.

Reply:
The issue is typescripts already a dev dependency but supporting versions from 2 through to 4, whereas this check has to use v5 because `@types` requires that as a min. version and I assume that the older versions are required for a reason.

I didn't create a tsconfig.json as that will be picked up by any running of typescript, and having it defined can change other aspects of linting such as what files are included by default.

Reply:
> shouldn't noEmit be set in tsconfig, and tsc be run on the whole project?

That'd mean the whole codebase would get type checked, requiring it to be correctly typed - while I personally would support that, I think it's out of scope for this pr

Reply:
So I did have a go at running `tsc` across the whole codebase just to see how much we'd get, and well with `strict` mode its 3498 errors and without it its 857 😅

Technically, we could have a `tsconfig.json` that includes just the `.d.ts` (and even `.js` because `checkJs` is false by default), but it feels weird to me having that as that only config option and for just a single small file...